### PR TITLE
add opt-in preact-refresh functionality

### PR DESCRIPTION
**Pull request by JoviDeCroock**
**What kind of change does this PR introduce?**
Feature

**Did you add tests for your changes?**
No, draft up until now will look into this later

**Summary**
This solves the need for hot module reloading within Preact, this is experimental and under ative d

**Does this PR introduce a breaking change?**
No, this is opt-in

**Other information**

```
Environment Info:
  System:
    OS: Windows 10 10.0.18362
    CPU: (16) x64 AMD Ryzen 7 2700X Eight-Core Processor
  Binaries:
    Node: 12.12.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.17.3 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.12.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 44.18362.449.0
  npmPackages:
    preact: ^10.3.2 => 10.4.1
    preact-cli: ^3.0.0-rc.6 => 3.0.0-rc.10
    preact-render-to-string: ^5.1.4 => 5.1.6
    preact-router: ^3.2.1 => 3.2.1
```

Sorry for the weird changes, my prettier seems to have over done it, will look into those later!

### CODE REVIEW

**Comment by prateekbh** [comment]
any reason why we want to put it behind the flag? 
Also what would be the indicator of "this is now ready to become default"

**Comment by JoviDeCroock** [comment]
@prateekbh I'm atm not sure about all scenario's and don't want to cause regressions. Now hooks aren't well supported and not entirely sure about error recov

File: packages/cli/lib/lib/webpack/webpack-client-config.js [hunk 1]
```diff
@@ -87,6 +90,9 @@ async function clientConfig(env) {
 		},
 
 		plugins: [
+			...(!isProd && refresh
+				? [new webpack.HotModuleReplacementPlugin(), new RefreshPlugin()]
```
Comment (prateekbh) [inline packages/cli/lib/lib/webpack/webpack-client-config.js:94]: are both of these required together?

Comment (JoviDeCroock) [reply packages/cli/lib/lib/webpack/webpack-client-config.js:94]: Yes `HotModuleReplacementPlugin ` enables `module.hot` to opt-out of page-reload

**Pull request merged**